### PR TITLE
NIO-859, implementing config specifications

### PR DESCRIPTION
**Pull request by f1401martin**

### CODE REVIEW

File: etc/logging.json [hunk 1]
```diff
@@ -42,7 +42,7 @@
     },
     "root": {
         "handlers": ["default", "file"],
-        "level": "WARNING"
+        "level": "INFO"
```
Comment (mattdodge) [inline etc/logging.json]: Think we probably want to keep the default log level in our logging config to `WARNING`. @hansmosh or @f1401martin can you confirm we don't want to change this?

File: nio.conf [hunk 2]
```diff
@@ -1,103 +1,276 @@
...
+#
+#master=True
+
+# the ip address of the master broker
+#
+broker_ip_address=[[COMHOST]]
+
+# your public ip address that the master broker can talk to you on
+#
+ip_address=0.0.0.0
```
Comment (mattdodge) [inline nio.conf:73]: Not sure if `0.0.0.0` will work for this setting, I think it needs to be an explicit IP address? Maybe `[[ IP_ADDR ]]` is better here?

File: nio.conf [hunk 3]
```diff
@@ -1,103 +1,276 @@
...
-[server]
 # settings for the web server that will host the core REST API
-host: [[NIOHOST]]
-port: [[NIOPORT]]
+#
+
+# host and port to launch rest web server on
+#
+port=[[NIOPORT]]
+host=[[NIOHOST]]
```
Comment (mattdodge) [inline nio.conf:231]: Should this be `0.0.0.0` by default? 

We have 3 different "IP address" environment variables, `NIOHOST`, `COMHOST`, and `PUBLIC_IP`. We probably don't need 3, but we definitely need them to be explicit in what their function is.

File: etc/logging.json (hunk 1 above)
Comment (f1401martin) [inline etc/logging.json]: Agree, should be kept at WARNING

File: nio.conf (hunk 2 above)
Comment (f1401martin) [inline nio.conf:73]: Just verified with a master and slave and it works fine keeping the setting at 0.0.0.0 (both instances running on the same computer)

File: nio.conf (hunk 3 above)
Comment (f1401martin) [inline nio.conf:231]: Could not find a reference to PUBLIC_IP (looks like a legacy setting)
Added comments to variables in nio.env

File: nio.conf (hunk 2 above)
Comment (f1401martin) [inline nio.conf:73]: Verified that when running instances on different computers this setting does not work

**Pull request merged**